src/tum_dlr_automl_for_eo/utils/helper_functions.py

```diff
@@ -0,0 +1,155 @@
+""""
+This file contains some constants and common functions to be used
+for LHC Sampling and to create {encoded_architecture:nb101_architecture} dictionary file.
+
+Following functions have been modified from https://github.com/kalifou/fitness_landscape_analysis_NAS
+
+"""
+from naslib.utils import nb101_api as api
+from pyDOE import *
+
+# Useful constants
+
+INPUT = 'input'
+OUTPUT = 'output'
+CONV3X3 = 'conv3x3-bn-relu'
+CONV1X1 = 'conv1x1-bn-relu'
+MAXPOOL3X3 = 'maxpool3x3'
+NUM_VERTICES = 7
+MAX_EDGES = 9
+EDGE_SPOTS = NUM_VERTICES * (NUM_VERTICES - 1) / 2  # Upper triangular matrix
+OP_SPOTS = NUM_VERTICES - 2  # Input/output vertices are fixed
+ALLOWED_OPS = [CONV3X3, CONV1X1, MAXPOOL3X3]
+ALLOWED_EDGES = [0, 1]  # Binary adjacency matrix
+
+CODING = ['input', 'conv1x1-bn-relu_0',
+          'conv1x1-bn-relu_1', 'conv1x1-bn-relu_2',
+          'conv1x1-bn-relu_3', 'conv1x1-bn-relu_4',
+          'conv3x3-bn-relu_0', 'conv3x3-bn-relu_1',
+          'conv3x3-bn-relu_2', 'conv3x3-bn-relu_3',
+          'conv3x3-bn-relu_4', 'maxpool3x3_0',
+          'maxpool3x3_1', 'maxpool3x3_2',
+          'maxpool3x3_3', 'maxpool3x3_4',
+          'output']
+
+
+def encoded_architecture_to_key(encoded_architecture):
+    """
+    @param encoded_architecture: encoded architecture array
+    @return key: encoded array as string to be used a key
+    """
+    key = ''.join([str(elem) for elem in encoded_architecture])
+    return key
+
+
+def key_to_encoded_architecture(architecture_key):
+    encoded_arc = [int(elem) for elem in architecture_key]
+    return encoded_arc
+
+
+def rename_ops(ops):
+    c1x1 = 0
+    c3x3 = 0
+    mp3x3 = 0
+    new_ops = []
+    for op in ops:
+        if op == CONV1X1:
+            new_ops = new_ops + [op + "_" + str(c1x1)]
+            c1x1 = c1x1 + 1
+        elif op == CONV3X3:
+            new_ops = new_ops + [op + "_" + str(c3x3)]
+            c3x3 = c3x3 + 1
+        elif op == MAXPOOL3X3:
+            new_ops = new_ops + [op + "_" + str(mp3x3)]
+            mp3x3 = mp3x3 + 1
+        else:
+            new_ops = new_ops + [op]
+    return new_ops
+
+
+def sample_single_op(NUM_VERTICES, ALLOWED_OPS):
+    x = lhs(NUM_VERTICES, samples=1)[0]
```

Comment:
I could not find the definition of the `lhc` function. Could you show it to me? @Stargazer1998

Reply:
Upps, I should have been more clear. It is a function from pyDOE lib.

Reply:
Hmm then we can import it as `import pyDOE` and use it like `pyDOE.lhs()`. 
Could you add the new required packages to the requirements.txt also?

Reply:
sure